test/unit/middlewares/tasks-validator.test.js

```diff
@@ -0,0 +1,349 @@
+const Sinon = require("sinon");
+const { getTasks } = require("../../../middlewares/validators/tasks");
+const { expect } = require("chai");
+const { TASK_STATUS } = require("../../../constants/tasks");
+
+describe("getTasks validator", function () {
+  it("should pass the request when no values for query params dev or status is passed", async function () {
+    const req = {};
+    const res = {};
+    const nextMiddlewareSpy = Sinon.spy();
+    await getTasks(req, res, nextMiddlewareSpy);
+    expect(nextMiddlewareSpy.callCount).to.be.equal(1);
+  });
+
+  it("should pass the request when dev query param value is boolean", async function () {
+    const req = {
+      query: {
+        dev: true,
+      },
+    };
+    const res = {};
+    const nextMiddlewareSpy = Sinon.spy();
+    await getTasks(req, res, nextMiddlewareSpy);
+    expect(nextMiddlewareSpy.callCount).to.be.equal(1);
+  });
+
+  it("should pass the request when dev query param has a boolean value of type string", async function () {
+    const req = {
+      query: {
+        dev: "true",
+      },
+    };
+    const res = {};
+    const nextMiddlewareSpy = Sinon.spy();
+    await getTasks(req, res, nextMiddlewareSpy);
+    expect(nextMiddlewareSpy.callCount).to.be.equal(1);
+  });
+
+  it("should not pass the request when dev query param value is not a boolean value", async function () {
+    const req = {
+      query: {
+        dev: "yes",
+      },
+    };
+    const res = {
+      boom: {
+        badRequest: Sinon.spy(),
+      },
+    };
+    const nextMiddlewareSpy = Sinon.spy();
+    await getTasks(req, res, nextMiddlewareSpy);
+    expect(nextMiddlewareSpy.callCount).to.be.equal(0);
```

Comment:
Nit : you can also expect `res.boom. badRequest.callCount` to equal 1

Reply:
Done